apps/fortuna/src/config.rs

```diff
@@ -158,3 +158,72 @@ pub struct EthereumConfig {
     /// The gas limit to use for entropy callback transactions.
     pub gas_limit: U256,
 }
+
+#[derive(Args, Clone, Debug)]
+#[command(next_help_heading = "Provider Config Options")]
+#[group(id = "ProviderConfig")]
+pub struct ProviderConfigOptions {
+    /// Path to a configuration file containing provider configuration for the list of supported blockchains
+    /// The commitments should be sorted in ascending order by their original_committed_sequence_number.
+    #[arg(long = "provider-config")]
+    #[arg(env = "FORTUNA_PROVIDER_CONFIG")]
+    #[arg(default_value = "provider-config.yaml")]
+    pub provider_config: String,
+}
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct ProviderConfig {
+    pub chains: HashMap<ChainId, ProviderChainConfig>,
+}
+
+impl ProviderConfig {
+    pub fn load(path: &str) -> Result<ProviderConfig> {
+        // Open and read the YAML file
+        let yaml_content = fs::read_to_string(path)?;
+        let config: ProviderConfig = serde_yaml::from_str(&yaml_content)?;
+        Ok(config)
+    }
+
+    /// Gets the provider chain config from the Hashmap. It will check that the commitments are in order.
+    /// If not, it will return an Err. Else, the config will be returned.
+    pub fn get_chain_config(&self, chain_id: &ChainId) -> Result<ProviderChainConfig> {
+        let provider_chain_config = self.chains.get(chain_id).map(|x| x.clone()).ok_or(anyhow!(
+            "Could not find chain id {} in the configuration",
+            &chain_id
+        ))?;
+
+        if provider_chain_config.check() {
+            return Ok(provider_chain_config);
+        }
+
+        Err(anyhow!(
+            "{}: error loading provider config. The commitments are not in order",
+            &chain_id
+        ))
+    }
+}
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct ProviderChainConfig {
+    pub commitments: Vec<Commitment>,
+}
+
+impl ProviderChainConfig {
+    pub fn check(&self) -> bool {
+        let commitments = &self.commitments;
+        for i in 0..commitments.len() - 1 {
+            if commitments[i].original_commitment_sequence_number
+                > commitments[i + 1].original_commitment_sequence_number
+            {
+                return false;
+            }
+        }
+        true
+    }
+}
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct Commitment {
+    pub metadata:                            String,
```

Comment:
let's use `seed` and `chain_length` for more observability about past commitments